docs/apps/components/checkbox.md

```diff
@@ -0,0 +1,198 @@
+        var fields= ['checkbox_g0x4pt54z']
+        this.hide(fields)
+        ```
+    
+    5. `this.show(fields)`
+        ```js
+        var fields= ['checkbox_g0x4pt54z']
+        this.show(fields)
+        ```
+    
+    6. `this.disable(['fields'])`
+        ```js
+        this.disable(['checkbox_g0x4pt54z'])
+        ```
+    
+    7. `this.enable(['fields'])`
+        ```js
+        this.enable(['checkbox_g0x4pt54z'])
+        ```
+    
+    8. `this.getValue('fieldName')`
+        ```js
+        var checkname = this.getValue('checkbox_g0x4pt54z');
+        console.log('getValue', checkname);
+        ```
+    
+    9. `this.getComponent('component_ID')`
+        ```js
+        var checkname = this.getComponent('checkbox_g0x4pt54z');
+        console.log('getComponent', checkname);
+        ```
+    
+    10. `this.setData(Value)`
+        ```js
+        this.setData(
+         {
+         "checkbox_g0x4pt54z": "Option 1", "Option 2",}
+        );
+
+!!! info Add multiple classes
+    Use the following syntax to add multiple classes to a component:
+    ```js
+    this.addClassName('componentID', 'class1', 'class2')
+    ```
+
+## Config
+
+| **Name**|**Description**|**Icon**|
+|---------|---------------|--------|
+|**ID**| The checkbox component's unique identifier|<img src= "/apps/components/img/input_id.png">|
+|**Name**| [**Optional**] The display name of the checkbox|<img src= "/apps/components/img/checkbox_name.png">|
+|**Width**| [**Optional**] Width of the field|<img src= "/apps/components/img/input_width.png">|
+|**Label Width**|Width of the label associated with an input field. It determines the horizontal space occupied by the label text|<img src= "/apps/components/img/input_labelwidth1.png">|
+|**Label Wrap**| If the label is longer than the allowed width the text will continue on another line|<img src= "/apps/components/img/input_labelwrap1.png">|
+|**Hide Label**| Hides the label on the form|<img src= "/apps/components/img/input_hidelabel.png">|
+|**Text Prompt**| A description to aid the user when completing the field|<img src= "/apps/components/img/input_textprompt.png">|
+|**Layout**| Select the format of the Checkbox as **Block** or **Inline**|<img src= "/apps/components/img/checkbox_layout.png">|
+|**Option**|Choose either **Static** or **Dynamic** way of adding data to the options of the Checkbox <br><br>**Static data** refers to a fixed set of options that are pre-defined and loaded into the Checkbox at the time of initialization</br></br> **Dynamic data** are options fetched as needed from sources like databases or APIs. The checkbox updates choices based on user selections. You select from **Data source** which refers to the origin of the data; **Function** which refers to the code that manipulates the data, and **Assigned Value** which refers to the specific value stored or associated with a data element|<img src= "/apps/components/img/checkbox_static.png"> <img src= "/apps/components/img/checkbox_dynamic.png">|
```

Comment:
The Dynamic Data bit is pretty confusing. Please clarify even more. It seems confusing even for me to use it effectively.

Reply:
modified the previous explanation as follows:

 **Dynamic data** option refers to the ability to automatically populate form/page fields or modify the form's/page behavior based on various sources like databases or APIs. The checkbox updates choices based on user selections. <br>You select from **Data source** takes information from databases, APIs etc; **Assigned Variable** refers to the specific value stored or associated with a data element and serves as a way to capture and manage dynamic data within the form/page and **Function** refers specific block of code that performs a particular task related to manipulating data and influencing the form's behavior. These functions operate on the data obtained from various sources like databases, APIs, user inputs and use them to dynamically modify the form experience.